Semantic-release workflow fails: no commits between main and release branch

## Summary

The semantic-release workflow is failing when attempting to create a release PR because no commits exist between main and the release branch.

## Error

Workflow run: https://github.com/Byte-Ventures/claude-trader/actions/runs/20314514278

```
pull request create failed: GraphQL: No commits between main and release/v1.33.0 (createPullRequest)
##[error]Failed to create PR for release 1.33.0
```

## Problem

The workflow successfully:
1. Calculates next version: `1.33.0`
2. Creates release branch: `release/v1.33.0`
3. Runs `semantic-release version --no-push --no-tag --no-vcs-release`
4. Pushes the release branch to remote

However, when it tries to create the PR:
```bash
gh pr create --base main --head "release/v1.33.0" ...
```

It fails because there are **no commits** between main and the release branch.

## Root Cause

The `semantic-release version` command with flags `--no-push --no-tag --no-vcs-release` appears to not be creating the expected commits that update version files (CHANGELOG.md, src/version.py, etc.).

Expected behavior:
- semantic-release should create commits updating:
  - `CHANGELOG.md` with release notes
  - `src/version.py` with new version number
  - Any other version-tracked files

Actual behavior:
- The release branch is created and pushed
- No commits are added to the branch
- Branch is identical to main

## Impact

- Releases cannot be created automatically
- Manual version bumping is required
- Release automation is completely broken

## Workflow Location

`.github/workflows/semantic-release.yml` - "Create release branch and PR" step

## Possible Causes

1. **semantic-release configuration issue** - The `--no-push` flag might be preventing commit creation
2. **Version file patterns not configured** - semantic-release might not know which files to update
3. **Dry-run mode** - semantic-release might be running in dry-run mode unintentionally
4. **Permission issue** - semantic-release might lack permissions to create commits

## Next Steps

1. Review `.python-semantic-release.toml` configuration
2. Test semantic-release locally with the same flags
3. Check if version file patterns are properly configured
4. Verify semantic-release is actually creating commits before push

## Priority

**High** - This blocks all automated releases.

--- a/.github/workflows/semantic-release.yml
+++ b/.github/workflows/semantic-release.yml
@@ -99,30 +99,46 @@ jobs:
           # Manually update version file
           # Note: semantic-release won't update files on non-release branches,
           # so we do it manually. The version was already calculated above.
-          # Use anchored regex to match only the actual assignment, not comments
           sed -i "s/^__version__ = .*/__version__ = \"${VERSION}\"/" src/version.py
+
+          # Validate the version was actually updated
+          if ! grep -q "^__version__ = \"${VERSION}\"" src/version.py; then
+            echo "::error::Failed to update src/version.py to version ${VERSION}"
+            cat src/version.py
+            exit 1
+          fi
 
           # Generate minimal changelog entry (full details in GitHub release)
           DATE=$(date +%Y-%m-%d)
           REPO_URL="${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}"
 
-          # Create changelog entry (no indentation to preserve markdown formatting)
           CHANGELOG_ENTRY="## v${VERSION} (${DATE})
 
 See [GitHub Release](${REPO_URL}/releases/tag/v${VERSION}) for details.
 "
 
-          # Update CHANGELOG.md
+          # Update CHANGELOG.md - insert before first version entry (## v)
           if [ -f CHANGELOG.md ] && [ -s CHANGELOG.md ]; then
-            # File exists and is not empty - insert after header
-            head -n 1 CHANGELOG.md > CHANGELOG.tmp
-            echo "" >> CHANGELOG.tmp
-            echo "$CHANGELOG_ENTRY" >> CHANGELOG.tmp
-            tail -n +2 CHANGELOG.md >> CHANGELOG.tmp
-            mv CHANGELOG.tmp CHANGELOG.md
+            if grep -q '^## v' CHANGELOG.md; then
+              # Insert before first version entry
+              awk -v entry="$CHANGELOG_ENTRY" '/^## v/ && !inserted {print entry; inserted=1} {print}' CHANGELOG.md > CHANGELOG.tmp
+              mv CHANGELOG.tmp CHANGELOG.md
+            else
+              # No version entries - append to end of file
+              echo "" >> CHANGELOG.md
+              echo "$CHANGELOG_ENTRY" >> CHANGELOG.md
+            fi
           else
-            # File doesn't exist or is empty - create fresh
-            echo -e "# Changelog\n\n${CHANGELOG_ENTRY}" > CHANGELOG.md
+            # File doesn't exist or is empty - create with standard header
+            cat > CHANGELOG.md << 'HEADER'
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+This file is automatically updated by semantic-release.
+
+HEADER
+            echo "$CHANGELOG_ENTRY" >> CHANGELOG.md
           fi
 
           # Stage changes and commit only if there are actual changes

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -8,9 +8,6 @@ build_command = ""
 match = "main"
 
 [tool.semantic_release.changelog]
-mode = "update"
-
-[tool.semantic_release.changelog.default_templates]
 changelog_file = "CHANGELOG.md"
 
 [tool.semantic_release.remote]

